checker.c

```diff
@@ -1,21 +1,164 @@
 #include <stdio.h>
 #include <assert.h>
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+/*------ module includes --------*/
+#include "checker.h"
+
+/*=============================================================================
+ =======               DEFINES & MACROS FOR GENERAL PURPOSE              =======
+ =============================================================================*/
+
+/*=============================================================================
+ =======                       CONSTANTS  &  TYPES                       =======
+ =============================================================================*/
+
+/*=============================================================================
+ =======                VARIABLES & MESSAGES & RESSOURCEN                =======
+ ==============================================================================*/
+static BMS_parameters_s BMS_parameters_attributes = {
+		FALSE,
+		FALSE,
+		FALSE,
+		FALSE
+};
+/*=============================================================================
+ =======                              METHODS                            =======
+ =============================================================================*/
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION: BMS_checkAttribute_OutOfRange
+ */
+/*!    \brief   Function to check the range of parameters
+ *
+ *
+ *     \param    	bms_attribute -> the parameter for which range needs to be checked
+					attribute_Min_Value -> Lower range of the parameter
+					attribute_Max_Value -> Upper range of the parameter
+					param[]	-> Parameter name to be printed
+ *     \returns  	void
+ *
+ *//*------------------------------------------------------------------------*/
+bool BMS_checkAttribute_OutOfRange(float bms_attribute, float attribute_Min_Value, float attribute_Max_Value, char param[])
+{
+	bool retAtributeStatus = FALSE;
+	if(bms_attribute < attribute_Min_Value || bms_attribute > attribute_Max_Value) 
+	{
+		BMS_printParameterStatus(param,"out of range!");
+		retAtributeStatus = TRUE;
+		BMS_setBMSStatus(TRUE);									
+	}	
+	
+	return retAtributeStatus;
+}
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION: BMS_printParameterStatus
+ */
+/*!    \brief   Function to print the parameter out of range
+ *
+ *
+ *     \param    	param[] -> the parameter will be printed
+ *     \returns  	void
+ *
+ *//*------------------------------------------------------------------------*/
+void BMS_printParameterStatus(char attribute[],char attribute_Status[])
```

Comment:
Accepting parameters by pointer: If you're not changing the content in the function, make it const, so the callers will know.
const not needed if you are accepting primitive types by value.